### PR TITLE
Bump `bitflags` to v2

### PR DESCRIPTION
**Pull request by daxpedda**
Changelog: https://github.com/bitflags/bitflags/blob/2.0.0/CHANGELOG.md#200.

This is a breaking change. The API that is implemented for bitflag types as well as the output by Serde is now different.

### CODE REVIEW

File: autogen/src/header.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ fn value_enum_attribute() -> TokenStream {
 
 fn bit_enum_attribute() -> TokenStream {
     quote! {
+        #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
```
Comment (MarijnS95) [inline autogen/src/header.rs:37]: even though these were the original `derive` attributes with `bitflags 1`, is `(Partial)Ord` really meaningful/useful for any of these?

Comment (daxpedda) [reply autogen/src/header.rs:37]: Probably not. They could be added on demand at a later point without a breaking change so I think there would be nothing wrong with removing them now.

Let me know!

Comment (MarijnS95) [reply autogen/src/header.rs:37]: Yeah, let's remove them. I double-checked all definitions and none of them seem to have any sort of meaning in ordering.

Comment (daxpedda) [reply autogen/src/header.rs]: Done.

**Review by msiglreith — APPROVED** [review]

**Comment by daxpedda** [comment]
Uh, I guess removing `Ord` had an issue after all.
`Decoration::FPFastMathMode` requires it. Should I remove it from `Decoration` or add it back to the macro again?

**Comment by msiglreith** [comment]
As `FPFastMathMode` is a bitmask with no defined partial/total order relation I would suggest removing `PartialOrd`/`Ord` from `Decoration` as well

**Comment by daxpedda** [comment]
Done.

**Review by MarijnS95 — APPROVED** [review]

**Review by msiglreith — APPROVED** [review]
Thanks, beside the test failure the other CI issues seem unrelated, I'm gonna merge it and fix the CI parts in a separate PR

**Pull request merged**